Makefile

```diff
@@ -1,59 +1,70 @@
 SHELL=/bin/bash
+export PATH := ./bin:./venv/bin:$(PATH)
+
 ifndef ELASTIC_VERSION
-ELASTIC_VERSION=5.3.0
+ELASTIC_VERSION := $(shell cat version.txt)
 endif
 
 ifdef STAGING_BUILD_NUM
-IMAGETAG=$(ELASTIC_VERSION)-${STAGING_BUILD_NUM}
-ES_DOWNLOAD_URL=http://staging.elastic.co/$(IMAGETAG)/downloads/elasticsearch
-ES_JAVA_OPTS:=ES_JAVA_OPTS="-Des.plugins.staging=${STAGING_BUILD_NUM}"
+VERSION_TAG := $(ELASTIC_VERSION)-$(STAGING_BUILD_NUM)
 else
-IMAGETAG=$(ELASTIC_VERSION)
-ES_DOWNLOAD_URL=https://artifacts.elastic.co/downloads/elasticsearch
+VERSION_TAG := $(ELASTIC_VERSION)
 endif
 
-ELASTIC_REGISTRY=docker.elastic.co
-VERSIONED_IMAGE=$(ELASTIC_REGISTRY)/elasticsearch/elasticsearch:$(IMAGETAG)
-LATEST_IMAGE=$(ELASTIC_REGISTRY)/elasticsearch/elasticsearch:latest
+ELASTIC_REGISTRY := docker.elastic.co
+VERSIONED_IMAGE := $(ELASTIC_REGISTRY)/elasticsearch/elasticsearch:$(VERSION_TAG)
 
-export ELASTIC_VERSION
-export ES_DOWNLOAD_URL
-export ES_JAVA_OPTS
-export VERSIONED_IMAGE
+# When invoking docker-compose, use an extra config fragment to map Elasticsearch's
+# listening port to the docker host.
+DOCKER_COMPOSE := 'docker-compose -f docker-compose.yml -f docker-compose.hostports.yml'
 
-.PHONY: build clean cluster-unicast-test pull-latest-baseimage push run-es-cluster run-es-single single-node-test test
+.PHONY: test clean pristine run run-single run-cluster build push
 
 # Default target, build *and* run tests
-test: single-node-test cluster-unicast-test
+test: lint build docker-compose.yml
+	./bin/testinfra tests
+
+lint: venv
+	flake8 tests
 
-# Clean up left over containers and volumes from earlier failed runs
 clean:
-	docker-compose down -v && docker-compose rm -f -v
+	docker-compose down -v
+	docker-compose rm -f -v
+	rm -f docker-compose.yml build/elasticsearch/Dockerfile
 
-run-es-single:
-	ES_NODE_COUNT=1 docker-compose -f docker-compose.yml -f docker-compose.hostports.yml up --build elasticsearch1
+pristine: clean
+	docker rmi -f $(VERSIONED_IMAGE)
 
-run-es-cluster:
-	ES_NODE_COUNT=2 docker-compose -f docker-compose.yml -f docker-compose.hostports.yml up --build elasticsearch1 elasticsearch2
+run: run-single
 
-single-node-test: export ES_NODE_COUNT=1
```

Comment:
Cool. Thanks for finding such a nice way to do this. My goal was to eliminate magic environment variables, not to make the tests less comprehensive!